src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -177,8 +177,7 @@ private void cancel() {
             listener().getLogger().println("Cancelling nested steps due to timeout");
             body.cancel(new ExceededTimeout());
             forcible = true;
-            long now = System.currentTimeMillis();
-            end = now + GRACE_PERIOD;
```

Comment:
Ok, the previous code would set `end = now + GRACE_PERIOD`, and then the call to `resetTimer` would set `end = now + timeout`, so this was effectively dead code and the grace period was the timeout.